lib/tapioca/compilers/dsl/active_record_associations.rb

```diff
@@ -249,6 +254,7 @@ def populate_collection_assoc_getter_setter(klass, constant, association_name, r
         end
         def type_for(constant, reflection)
           return "T.untyped" if !constant.table_exists? || polymorphic_association?(reflection)
+          raise MissingConstantError unless Object.const_defined?(reflection.class_name)
```

Comment:
Can we use the same mechanism as Active Record to do the `const_defined?` check? If Rails ever changes how it does that internally, we might end up not matching in behaviour. That's why I'd suggested `safe_constantize` instead:
```suggestion
          raise MissingConstantError unless reflection.class_name.safe_constantize
```